rust/arrow/src/json/reader.rs

```diff
@@ -0,0 +1,853 @@
+                                values.insert(k.to_string(), hs);
+                            }
+                        }
+                        Value::Bool(_) => {
+                            if values.contains_key(k) {
+                                let x = values.get_mut(k).unwrap();
+                                x.insert(DataType::Boolean);
+                            } else {
+                                // create hashset and add value type
+                                let mut hs = HashSet::new();
+                                hs.insert(DataType::Boolean);
+                                values.insert(k.to_string(), hs);
+                            }
+                        }
+                        Value::Null => {
+                            // do nothing, we treat json as nullable by default when
+                            // inferring
+                        }
+                        Value::Number(n) => {
+                            if n.is_f64() {
+                                if values.contains_key(k) {
+                                    let x = values.get_mut(k).unwrap();
+                                    x.insert(DataType::Float64);
+                                } else {
+                                    // create hashset and add value type
+                                    let mut hs = HashSet::new();
+                                    hs.insert(DataType::Float64);
+                                    values.insert(k.to_string(), hs);
+                                }
+                            } else {
+                                // default to i64
+                                if values.contains_key(k) {
+                                    let x = values.get_mut(k).unwrap();
+                                    x.insert(DataType::Int64);
+                                } else {
+                                    // create hashset and add value type
+                                    let mut hs = HashSet::new();
+                                    hs.insert(DataType::Int64);
+                                    values.insert(k.to_string(), hs);
+                                }
+                            }
+                        }
+                        Value::String(_) => {
+                            if values.contains_key(k) {
+                                let x = values.get_mut(k).unwrap();
+                                x.insert(DataType::Utf8);
+                            } else {
+                                // create hashset and add value type
+                                let mut hs = HashSet::new();
+                                hs.insert(DataType::Utf8);
+                                values.insert(k.to_string(), hs);
+                            }
+                        }
+                        Value::Object(_) => {
+                            panic!("Reading nested JSON structes currently not supported")
```

Comment:
Can we make this return `Result` instead of panicking

Reply:
Done